src/lib.rs

```diff
@@ -99,6 +99,107 @@ impl MetalIrConverter {
     }
 }
 
+macro_rules! versioned_info {
+    ($name:ident, $create:ident, $release:ident) => {
+        pub struct $name {
+            me: ffi::$name,
+            funcs: Arc<bindings::metal_irconverter>,
+        }
+
+        impl Deref for $name {
+            type Target = ffi::$name;
+
+            fn deref(&self) -> &Self::Target {
+                &self.me
+            }
+        }
+
+        impl Drop for $name {
+            fn drop(&mut self) {
+                assert!(unsafe { self.funcs.$release(&mut self.me) })
+            }
+        }
+
+        impl $name {
+            unsafe fn new(
```

Comment:
Can we reduce the scope of `unsafe` by not marking the function as `unsafe`? This behaviour is also changing in Rust edition 2024, and it'd be nice to preempt that.

Reply:
In other words, write the function exactly as the original `fn compute_info()`, but with the macro expansion.

Reply:
Updated